### PR TITLE
Ren and Cleo modal fixes

### PR DESCRIPTION
**Pull request by Benji42**
Now, Ren and Cleo both have a modal to select which attack to copy. The problem with the hooks being deleted every turn has a also been fixed, so they're only removed when the mimicked card changes.

In addition, X's modal was updated to be more consistent with the other two.

The modal requests have an option to supply a picked slot now, to work with Ren and Cleo, which seemed like the best solution to me. However I'm not totally sure (tbf modal requests are still messy in general)

### CODE REVIEW

**Comment by Sense101** [comment]
This ready to merge now?

**Comment by Benji42** [comment]
yes

File: client/src/logic/game/tasks/action-logic-saga.ts [hunk 1]
```diff
@@ -42,18 +42,18 @@ function* actionLogicSaga(gameState: LocalGameState): SagaIterator {
 	const pState = gameState.players[playerId]
 	const lastActionResult = gameState.lastActionResult
 
-	if (gameState.currentModalId !== null) {
-		const id = gameState.currentModalId
+	if (gameState.currentModalData && gameState.currentModalData.modalId) {
+		const id = gameState.currentModalData?.modalId
 		if (id === 'grian_rare') {
 			yield fork(borrowSaga)
```
Comment (Sense101) [inline client/src/logic/game/tasks/action-logic-saga.ts:48]: One more thing, actually - can you see how here a huge portion of this if/else is now meaningless? Unless if it's grian, for which we have a custom saga, it should just be setOpenedModal(id)

**Review by Sense101 — CHANGES_REQUESTED** [review]

File: client/src/logic/game/tasks/action-logic-saga.ts (hunk 1 above)
Comment (Benji42) [reply client/src/logic/game/tasks/action-logic-saga.ts:48]: i'll take a look

Comment (Benji42) [reply client/src/logic/game/tasks/action-logic-saga.ts:48]: cool fixed

**Review by Sense101 — APPROVED** [review]

**Pull request merged**